### PR TITLE
[DF] Teach `Vary` to accept `Stats`

### PR DESCRIPTION
**Pull request by ikabadzhov**
With this solution `TStatistic` never resets its result that is being varied.

Fix https://github.com/root-project/root/issues/10545.

### CODE REVIEW

**Comment by ikabadzhov** [comment]
@phsft-bot build just on ROOT-debian10-i386/cxx14

File: tree/dataframe/test/dataframe_vary.cxx [hunk 1]
```diff
@@ -654,6 +654,27 @@ TEST_P(RDFVary, VariedHistosMustHaveNoDirectory)
...
+   EXPECT_EQ(hs1["x:1"].GetMean(), 2);
+   auto hs2 = VariationsFor(h);
+   EXPECT_EQ(hs2["x:0"].GetMean(), -1);
+   EXPECT_EQ(hs2["x:1"].GetMean(), 2);
+
+   auto s = df.Vary("x", SimpleVariation, {}, 2).Stats<int>("x");
+   auto ss1 = VariationsFor(h);
+   EXPECT_EQ(ss1["x:0"].GetMean(), -1);
+   EXPECT_EQ(ss1["x:1"].GetMean(), 2);
+   auto ss2 = VariationsFor(h);
```
Comment (eguiraud) [inline tree/dataframe/test/dataframe_vary.cxx:673]:
```suggestion
   auto ss2 = VariationsFor(s);
```

**Review by eguiraud — CHANGES_REQUESTED** [review]

File: tree/dataframe/test/dataframe_vary.cxx [hunk 2]
```diff
@@ -654,6 +654,27 @@ TEST_P(RDFVary, VariedHistosMustHaveNoDirectory)
...
+   EXPECT_EQ(ss2["x:0"].GetMean(), -1);
+   EXPECT_EQ(ss2["x:1"].GetMean(), 2);
+}
```
[the remark this answers is not in the document]
Comment (ikabadzhov) [reply tree/dataframe/test/dataframe_vary.cxx:676]: It does fail. I just checked. The exact same test fails  with:
```
394: [ ]/tree/dataframe/test/dataframe_vary.cxx:667: Failure
394: Expected equality of these values:
394:   hs2["x:0"].GetMean()
394:     Which is: 0.6
394:   -1
394: [ ]/tree/dataframe/test/dataframe_vary.cxx:668: Failure
394: Expected equality of these values:
394:   hs2["x:1"].GetMean()
394:     Which is: 1.2
394:   2
```

File: tree/dataframe/test/dataframe_vary.cxx [hunk 3]
```diff
@@ -654,6 +654,27 @@ TEST_P(RDFVary, VariedHistosMustHaveNoDirectory)
    EXPECT_EQ(hs["x:1"].GetDirectory(), nullptr);
 }
 
+TEST_P(RDFVary, FillHelperResets)
```
[the remark this answers is not in the document]
Comment (ikabadzhov) [reply tree/dataframe/test/dataframe_vary.cxx:658]: Only this remark remained open now: I added:
```
// check if the Fill action helper properly resets between event loops
```

**Review by eguiraud — APPROVED** [review]
Thanks Ivan, nice solution! Just a wording suggestion for your consideration, but this can go in.

**Pull request merged**